test/functional/test_framework/mininode.py

```diff
@@ -241,15 +258,19 @@ def _log_message(self, direction, msg):
         logger.debug(log_message)
 
 
-class NodeConnCB():
-    """Callback and helper functions for P2P connection to a bitcoind node.
+class NodeConnCB(NodeConn):
+    """A high-level P2P interface class for communicating with a Bitcoin node.
+
+    This class provides high-level callbacks for processing P2P message
+    payloads, as well as convenience methods for interacting with the
+    node over P2P.
 
     Individual testcases should subclass this and override the on_* methods
-    if they want to alter message handling behaviour."""
-    def __init__(self):
-        # Track whether we have a P2P connection open to the node
-        self.connected = False
-        self.connection = None
+    if they want to alter message handling behaviour.
+
+    TODO: rename this class P2PInterface"""
+    def peer_connect(self, *args, **kwargs):
+        super().peer_connect(*args, **kwargs)
 
         # Track number of messages of each type received and the most recent
```

Comment:
In commit "[tests] Make NodeConnCB a subclass of NodeConn"

Maybe keep assigning nServices/messages_count/last_message/ping_counter members that don't depend on connection information in `__init__` rather than `peer_connect`. This would mean you don't need to introduce a `NodeConnCB.peer_connect` method in this commit, and in the next commit it can be a shorter, more focused override. Just a suggestion, though, difference is minor.

Reply:
Agree. Makes more sense. Thanks.